experimental/dds/tree2/src/core/undo/undoRedoManager.ts

```diff
@@ -181,36 +179,30 @@ export class UndoRedoManager<TChange, TEditor extends ChangeFamilyEditor> {
 
 	/**
 	 * Updates the state of this {@link UndoRedoManager} to correctly reference commits that have been rebased after merging.
-	 * @param newHead - the head commit of the newly rebased branch.
+	 * @param newCommits - All commits which were not part of the source branch before the rebase, but are after the rebase.
 	 * @param mergedUndoRedoManager - the {@link UndoRedoManager} of the branch that was merged.
 	 */
 	public updateAfterMerge(
-		newHead: GraphCommit<TChange>,
+		newCommits: GraphCommit<TChange>[],
 		mergedUndoRedoManager: UndoRedoManager<TChange, TEditor>,
 	): void {
-		if (this.getHead !== undefined) {
-			this.updateBasedOnNewCommits(this.getHead(), newHead, this, mergedUndoRedoManager);
-		}
+		this.updateBasedOnNewCommits(newCommits, this, mergedUndoRedoManager);
 	}
 
 	/**
 	 * Updates the state of this {@link UndoRedoManager} to correctly reference commits that have been rebased.
-	 * @param newHead - the head commit of the newly rebased branch.
+	 * @param newCommits - All commits which were not part of the source branch before the rebase, but are after the rebase.
```

Comment:
"all commits from the original branch that have rebased versions present on the new branch" or something like that